mne/viz/decoding.py

```diff
@@ -129,17 +145,35 @@ def plot_gat_diagonal(gat, title=None, xmin=None, xmax=None, ymin=0., ymax=1.,
     import matplotlib.pyplot as plt
     if ax is None:
         fig, ax = plt.subplots(1, 1)
-    # detect whether gat is a full matrix or just its diagonal
-    if np.all(np.unique([len(t) for t in gat.test_times_['times_']]) == 1):
-        scores = gat.scores_
+
+    # Find and plot chance level
+    if chance is not False:
+        if chance is True:
+            chance = _get_chance_level(gat.scorer_, gat.y_train_)
+        elif type(chance) not in [int, float, np.float64, np.float32]:
+            raise ValueError('\'chance\' must be int or float')
```

Comment:
this is not necessary. Just do:

chance = float(chance)

it will crash if chance cannot be cast to float